frontend/src/metabase/querying/notebook/components/NotebookDataPicker/EmbeddingDataPicker/EmbeddingDataPicker.tsx

```diff
@@ -74,14 +89,22 @@ export function EmbeddingDataPicker({
     );
   }
 
+  const isSourceSelected = Boolean(pickerInfo?.tableId);
   return (
     <PLUGIN_EMBEDDING.DataSourceSelector
-      key={pickerInfo?.tableId}
-      isInitiallyOpen={!table}
+      key={
+        isSourceSelected
+          ? pickerInfo?.tableId
+          : `${sourceTable?.id}:${isSourceModelFetching}`
+      }
+      isInitiallyOpen={isSourceModelFetching ? false : !table}
+      isQuerySourceModel={isSourceModel}
```

Comment:
this prop will be used to choose between opening the model or a table in the join step.